.github/workflows/check-code-samples.sh

```diff
@@ -0,0 +1,75 @@
+cd ./code-samples/
+files=$(ls | wc -l)
+echo "Check $files files …"
+failedFiles=()
+notRunnable=0
+i=0
+for file in *.pony; do # actors-sequential.pony
+    ((i++))
+    percentage=$(((i*100)/files))
+    echo -e "#$i Test $file … ($i/$files \u2192 $percentage %)"
+    contents=$(cat "$file") # code-samples/
+    #echo "contents: $contents"
+    regex='^"""\n(?<line>^(?<key>[a-z_]+):\s?(?<value>.*?)$\n)+"""' # TODO: Get frontmatter from AST
+    regex='^"""\n(.*)\n"""'
+    expectations=$(jq --arg file "${file}" ".[\"$file\"]" ../code-samples.json) # -r
+    #echo "Expectations for $file $expectations"
+    isRunnable=$(echo "$expectations" | jq '.runnable')
+    if ! [ -z "$expectations" ] && ! $isRunnable; then
+        echo -e "\u2139\uFE0F File not runnable. Skip"
+        ((notRunnable++))
+        continue
+    fi
+    expectedStdout=$(echo "$expectations" | jq '.stdout')
+    expectedStderr=$(echo "$expectations" | jq '.stderr')
+    expectedExitcode=$(echo "$expectations" | jq '.exitcode')
+    #echo "stdout for $file: "
+    #echo "$contents" | grep -e $regex
+    if [ $(echo "$contents" | grep -qe $regex) ]; then # TODO: fix regex
+        frontmatter="${BASH_REMATCH[0]}"
+        readarray -d "\n" -t lines <<< frontmatter
+        echo "Expectations 1: ${lines[0]}"
+    fi
+    if [[ "$contents" =~ $regex ]]; then
+        frontmatter="${BASH_REMATCH[0]}"
+        readarray -d "\n" -t lines <<< frontmatter
+        echo "Expectations: ${lines[0]}"
+    fi
+    encoded=$(sed -E -e ':a;N;$!ba;s/\r\n|\r|\n/\\n/g' "$file") #code-samples/
+    encoded=${encoded//\"/\\\"}
+    encoded=${encoded//   /\\t}
+    json="{\"code\": \"$encoded\", \"separate_output\": true, \"color\": true, \"branch\": \"release\"}"
+    #echo "Send $json…"
+    response=$(curl -s --header "Content-Type: application/json" \
+        --request POST \
+        --data "$json" \
+        "https://playground.ponylang.io/evaluate.json")
```

Comment:
Rather than calling out to the playground to run these, we should use `ponyc` directly here. We don't want to flood the playground when running these tests, and we don't want the uptime of the playground to affect these tests.

Reply:
Well, I implemented it in a way I'm familiar, since nobody replied to "I still need to figure out, how that might work 🤔 If there's someone else who would take this on, I wouldn't be mad 😅" in #340. I can try doing it via `ponyc` but how it will turn out is up to be shown.